velox/docs/functions/presto/decimal.rst

```diff
@@ -127,9 +127,42 @@ difference in s1 and s.
 
 	p = p1 + s2 + max(0, s2 - s1)
 
-Like in Addition, the precision of the result may exceed 38. The choices are the
-same. Presto chooses to cap p at 38 and allow runtime errors.
+Let’s say :code:`num1 is a decimal(p1, s1)` with unscaled value `A` and :code:`num2 is a decimal(p2, s2)`
+with unscaled value B.
 
+::
+
+    num1 = A / 10^s1
+    num2 = B / 10^s2
+
+The result type precision and scale are:
+
+::
+
+	s = max(s1, s2)
+	p = p1 + s2 + max(0, s2 - s1)
+
+The final result:
+::
+
+    r = A * 10^(s + s2 - s1) / B
+
+To compute this value, we first rescale A using the rescale factor (s + s2 - s1), then divide by B and
+round to the nearest whole. This method works as long as the :code:`rescale factor < 38`. If :code:`s + s2 - s1`
+exceeds 38 an error is raised.
+
+**Result scale and rescale factor**
+
+Let's understand why result scale is :code:`max(s1,s2)`. This is crucial in computing rescaling factor (rf).
```

Comment:
This explanation is confusing to me. It reads as if result scale is chosen based on a specific implementation, which in turn was chosen somewhat arbitrary.

`This is crucial in computing rescaling factor (rf).`

Also, an example is helpful to illustrate a generic point or provide an intuition for coming up with a generic solution. An example cannot be used to explain a generic point.

I'm not sure I understand why Presto chose max(s1, s2) as result scale. It could choose s1 as well (or any other value). The result of a division (unlike other operations) is approximate, hence, one is free to decide how accurate it needs to be. 

Perhaps, in Presto the thinking was to assume that user's desired accuracy is the max of input scales. However, one could also say that desired accuracy is the scale of the dividend. In SQL, literal values get their types assigned by the actual number of digits after the decimal point. Hence, in the following SQL 1.2 has scale 1 and 0.01 has scale 2. One may argue that user's intention is to work with accuracy of 2 digits after the decimal point, hence, max(s1, s2).

SELECT 1.2 / 0.01

If this makes sense, maybe we could replace "It is not clear exactly why." with this reasoning (making it clear that this is a speculation, not a definitive reason).